### ISSUE TITLE
Training Issue:All Epoch Weights Zeroed Except epoch-50.pth

### ISSUE THREAD

**Issue by Yunmu-deep** [kimchange/SeReNet#2]
Dear Author,

Hello, we encountered some issues while training the SeReNet network. We trained SeReNet using the synthetic Bubtub dataset from the folder by running the train_serenet.py file. After approximately 17 hours, we obtained results stored in the 202506 folder.

We used the following code to inspect the parameters in each .pth file:

import torch
import pprint
weights = torch.load('E:/SeReNet-main/save/202506/serenet_Nnum13_bubtub/epoch-50.pth')
print("Model parameters:")
pprint.pprint(weights)

We observed that：
1.Only epoch-50.pth contains non-zero parameters
2.All other epoch-××.pth files contain model parameters where every single parameter is exactly zero (completely zero-valued)

Aside from necessary path modifications, we made no other changes to the code. We would greatly appreciate your insights into this situation.Attached are the parameter results of the SeReNet model we trained.

[202506.zip](https://github.com/user-attachments/files/20879110/202506.zip)

Thank you for your time and assistance.

**Comment by kimchange** [comment]
Below is a training history by one undergraduate last month, I think this may help you. (Theoretically, loss numbers would be nearly the same, except for accuracy mismatches caused by parallel computing order).
[serenet_Nnum13_bubtub_lpips_20250530.zip](https://github.com/user-attachments/files/20886788/serenet_Nnum13_bubtub_lpips_20250530.zip)

From your upload files, I find the only difference in "log.txt" is that, "\\" appeared, seems that you are using windows, while I haven't a windows machine equiped with GPU. So is there a "code" backup in your savefolder ? That is more helpful for me to debug.

I just find that, before epoch 50, everything seems just OK, the loss decreased to about 0.07, that is the same to my uploaded record in this answer. But in epoch 51, it crushed. I think you can first, simply modify config file:
epoch_val: 50
--->
epoch_val: 99999
Then finish training, see what will happen to your loss curve. (I think maybe there is something wrong with validation process in your machine, this modification will jump all vals ) 

Waiting for your results !

**Comment by Yunmu-deep** [comment]
Thank you very much for your prompt response. Following your suggestion, we modified epoch_val: 50 to epoch_val: 99999 to skip the validation phase. The results now closely align with those you provided, and the issue of model parameters becoming zero after epoch 50 no longer occurs.

Subsequently, we ran the test_serenet.py script using the resulting epoch_800.pth file. The test results are consistent with those obtained from ../SeReNet-main/demo/pth/serenet_pth/epoch_800.pth. This confirms that the problem truly lies in the validation workflow, which appears to cause abrupt disruptions during model training.

Regarding the validation process, do you have any recommendations to resolve this issue?

Attached for your reference are:

1.The modified configuration files (with adjusted epoch_val values) used in our training.
2.The training results generated after implementing these changes.
And we are using windows.

We sincerely appreciate your advice and assistance.

[save.zip](https://github.com/user-attachments/files/20915136/save.zip)

**Comment by kimchange** [comment]
Try to modify line 37-38 in "save/month_202505/config_tag/code/train_serenet.py" (or other dataloaders) in windows:
```
loader = DataLoader(dataset, batch_size=spec['batch_size'],
        shuffle=(tag == 'train'), num_workers=1+5*int(tag == 'train'), pin_memory=True, persistent_workers=True)
```
to
```
loader = DataLoader(dataset, batch_size=spec['batch_size'],
        shuffle=(tag == 'train'), num_workers=0, pin_memory=True)
```

**Comment by Yunmu-deep** [comment]
Thanks for the effective and timely advice, the problem has now been resolved.
